### ISSUE TITLE
Add push notifications

### ISSUE THREAD

**Issue by D-Marc1** [WebsiteBeaver/CordovaCall#5]
- [ ] PushKit
- [ ] Regular Push Notifications for Android

**Comment by jQrgen** [comment]
This would be awesome, would we support silent push notifications that wake up background jobs?

**Comment by D-Marc1** [comment]
Ya this is likely all it would do. For iOS, PushKit was specifically made for this purpose; for Android it would just be a silent notification probably. There is no timeline for any of the new features, but it's largely predicated on how many stars this repo gets. If it gets popular enough, I know that @dmarcs would be happy to add stuff like this.

**Comment by jQrgen** [comment]
Great,

How does the plugin fit into an architecture now? Do you have to use a seperate plugin for VOIP pushed which triggers this plugin in background on a VOIP push?

Thanks

**Comment by D-Marc1** [comment]
Yep, for now just use a separate plugin for PushKit to wakeup the app; that's all you really need to do.

**Comment by ostrichegret** [comment]
@D-Marc1 hi, do you have a recommendation of cordova plugin for showing call screen like whatsapp when the phone screen is off and lock? 

I can only find push notification that show on notification bar and need user click to open the app.

Thank you

**Comment by D-Marc1** [comment]
You are literally describing this plugin. For PushKit, use [this plugin](https://github.com/Hitman666/cordova-ios-voip-push).

**Comment by Bobisback** [comment]
@D-Marc1 Is there a guide for android to do something similar to pushkit? Or do you just send a silent notification and then call cordovacall?

**Comment by mattkhaw** [comment]
@D-Marc1 I'm just gonna comment here on how to get both iOS and Android to work with push notifications. I hope this will help and I think the plugin doesn't require to be modified for Android, but for iOS definitely yes since I've encountered this issue before and I've made a working plugin that uses PushKit with CordovaCall.

For iOS, there's no way to get it working separately since iOS 13 (you still can getaway with prior versions). Just use [this](https://github.com/mattkhaw/cordova-plugin-callkit) instead.

For Android, just use background notifications. The reason why I think this plugin doesn't need any modifications on this end is because of the variety of push notifications libraries. Some might use OneSignal and some might use the standard FCM as an example. Therefore, there's no need for that.
